lib/rubocop/cop/rspec/empty_hook.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      #  Checks for empty before and after hooks.
+      #
+      # @example
+      #   # bad
+      #   before {}
+      #   after do; end
+      #   before(:all) do
+      #   end
+      #   after(:all) { }
+      #
+      #   # good
+      #   before { create_users }
+      #   after do
+      #     cleanup_users
+      #   end
+      #   before(:all) do
+      #     create_feed
+      #   end
+      #   after(:all) { cleanup_feed }
+      class EmptyHook < Cop
+        include RuboCop::Cop::RangeHelp
+
+        MSG = 'Empty hook detected.'
+
+        def_node_matcher :empty_hook?, <<~PATTERN
+          (block $#{Hooks::ALL.send_pattern} _ nil?)
+        PATTERN
+
+        def on_block(node)
+          empty_hook?(node) do |hook|
+            add_offense(hook)
+          end
+        end
+
+        def autocorrect(node)
+          lambda do |corrector|
+            block = node.parent
+            range = range_by_whole_lines(block.loc.expression,
```

Comment:
In the unlikely event that anyone has `before {} ; after { clean_up(:foo) }` on the same line, the autocorrect would remove both. Maybe `#range_with_surrounding_space` is safer to use here?

I wonder how many coops are using `#range_by_whole_lines` and don’t work with multiple statements on the same line? 🤷‍♂️

Reply:
This is a little tricky. `range_with_surrounding_space` does not remove the semi-colon. We probably need to add method like `range_with_semicolon` to `RangeHelp`.

Reply:
Removing the semicolon is a job for the `Style/Semicolon` cop.

In our specs you’ll sometimes see an autocorrection introducing e.g. [a blank line at the end of a block](https://github.com/rubocop-hq/rubocop-rspec/blob/0f7721ed39c2276a8f8d7a796213b19ed6bf6b6a/spec/rubocop/cop/rspec/scattered_let_spec.rb#L71-L84) or similar “ugly” formatting. And that’s fine. When auto-correcting RuboCop will automatically run the resulting file though the cops once more to see if new offenses were introduced.

Reply:
I have this solution that passes the test - but is quite ugly (hence not committed):

```ruby
        def autocorrect(node)
          lambda do |corrector|
            corrector.remove(range_for_autocorrect(node))
          end
        end

        private

        def range_for_autocorrect(node)
          block = node.parent
          range = block.loc.expression
          buffer = @processed_source.buffer
          # Handle multiple statements on same line
          if buffer.source[range.end_pos] == ';'
            range = Parser::Source::Range.new(@processed_source.buffer,
                                              range.begin_pos,
                                              range.end_pos + 1)
          end
          range_with_surrounding_space(range: range)
        end
```

Reply:
That's neat 👍 I've changed autocorrect to use `range_with_surrounding_space`